### PR TITLE
Use ExceptT correctly instead of throwIO.

### PR DESCRIPTION
**Pull request by ksaric**

### CODE REVIEW

File: cardano-node/src/Cardano/CLI/Delegation.hs [hunk 1]
```diff
@@ -58,15 +61,15 @@ checkByronGenesisDelegation
...
   -> Crypto.VerificationKey
   -> Crypto.VerificationKey
-  -> IO ()
+  -> ExceptT CliError IO ()
 checkByronGenesisDelegation (CertificateFile certF) magic issuer delegate = do
-  ecert <- canonicalDecodePretty <$> LB.readFile certF
+  ecert <- liftIO $ canonicalDecodePretty <$> LB.readFile certF
   case ecert of
-    Left e -> throwIO $ DlgCertificateDeserialisationFailed certF e
+    Left e -> left $ DlgCertificateDeserialisationFailed certF e
```
Comment (mrBliss) [inline cardano-node/src/Cardano/CLI/Delegation.hs]: I prefer `throwError` to `left` :slightly_smiling_face:

File: cardano-node/src/Cardano/CLI/Delegation.hs [hunk 2]
```diff
@@ -14,6 +14,9 @@ module Cardano.CLI.Delegation
 where
 
 import           Cardano.Prelude hiding (option, show, trace)
+
+import           Control.Monad.Trans.Except.Extra (left)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/CLI/Delegation.hs]: `import           Control.Monad.Trans.Except.Extra (left)` should be moved as follows:
```
import           Control.Monad.Trans.Except.Extra (left)
import qualified Data.ByteString.Lazy as LB
```

File: cardano-node/src/Cardano/CLI/Delegation.hs (hunk 1 above)
Comment (erikd) [reply cardano-node/src/Cardano/CLI/Delegation.hs]: I prefer `left` to `throwError`. The former suggests a connection to the `Left` of an `Either` whereas the later uses "throw' which (incorrectly) suggests a connection to exceptions. `

File: cardano-node/src/Cardano/CLI/Tx.hs [hunk 3]
```diff
@@ -80,11 +80,11 @@ prettyAddress addr = sformat
   (Common.addressF %"\n"%Common.addressDetailedF)
   addr addr
 
-readByronTx :: TxFile -> IO (GenTx ByronBlock)
+readByronTx :: TxFile -> ExceptT CliError IO (GenTx ByronBlock)
 readByronTx (TxFile fp) = do
-  txBS <- LB.readFile fp
+  txBS <- liftIO $ LB.readFile fp
   case fromCborTxAux txBS of
```
Comment (erikd) [inline cardano-node/src/Cardano/CLI/Tx.hs]: Maybe [hoistEither](http://hackage.haskell.org/package/transformers-except-0.1.1/docs/Control-Monad-Trans-Except-Extra.html) can make that even nicer.

**Pull request merged**